truffle/src/com.oracle.truffle.polyglot/src/com/oracle/truffle/polyglot/PolyglotContextImpl.java

```diff
@@ -932,6 +940,12 @@ public boolean initializeLanguage(String languageId) {
         return false;
     }
 
+    @Override
+    public boolean initializeLanguage(String languageId) {
+        PolyglotLanguage language = requirePublicLanguage(languageId);
+        return initializeLanguage(language);
```

Comment:
Use this implementation instead. The PolyglotImpl.guestToHostException method should only be used for calls from the embedding API and not from the TruffleContext.

```
 @Override
    public boolean initializeLanguage(String languageId) {
        PolyglotLanguage language = requirePublicLanguage(languageId);
        PolyglotLanguageContext languageContext = getContext(language);
        Object prev = hostEnter(languageContext);
         try {
              return initializeLanguage(language);
         } catch (Throwable t) {
            throw PolyglotImpl.guestToHostException(languageContext, t, true);
        } finally {
            hostLeave(languageContext, prev);
        }
   }
```